src/Helpers/WalletParser.cs

```diff
@@ -0,0 +1,108 @@
+using NBitcoin;
+using NBitcoin.Scripting;
+using NBXplorer.DerivationStrategy;
+
+namespace NodeGuard.Helpers;
+
+public static class WalletParser
+{
+    /// <summary>
+    /// Parse the output descriptor string to get the wallet info, Took from BTCPAYServer codebase
+    /// </summary>
+    /// <param name="outputDescriptorStr"></param>
+    public static (DerivationStrategyBase, (BitcoinExtPubKey, RootedKeyPath)[]) ParseOutputDescriptor(
```

Comment:
Cannot this be returned in a more structure way than nested tuples?

Reply:
Tuples are easier to maintain than classes 🤔

Reply:
But they are way less readable, when you get the result you need to play with: `result.Item1` and so on which make it not readable enough 🤔  and makes you check the signature of the function to see actually what you are receiving